Add "avoiding disjointed features syndrome" (#20)
RaphHaddad/blog: content/post/avoiding-disjointed-features-syndrome.md

---
title: 'Avoiding the Disjointed Features Syndrome'
date: 2020-05-20T00:00:00.000+10:00
draft: false
tags : [agile]
---

The "disjointed features syndrome" is a phenomenon that occurs when features
are built with the resulting product being a collection of features that may
not integrate correctly, have inconsistent user interfaces, and/or user
experiences.

For example, the below table has the following features:

- It has four legs; and
- Is able to be placed on a flat surface.

![](/images/table.png)

Although the table above satisfies all the requirements, it's still an
unusable product. Another example is a disjointed document where paragraphs
aren't orthographically consistent; such as the usage of different tense or
points of view (among other things).

There are several reason why the disjointed features syndrome may occur,

> below is an outline of a few delivery behaviours to avoid to help

Author: will fix.

avoid this syndrome.

## Feature-Only Review Ceremonies

All too often review ceremony meetings become only about the last iteration.
When this happens, the only changes that are examined are the features completed
in the last iteration itself, instead of the entire product end-to-end.

This creates a situation where the focus is on new features, not how they
interact with the entire product. More often than not, it isn't new features
that make a product valuable, rather, it's the removal of unnecessary features
that enable this visibility of value in a focussed fashion.

To avoid this behaviour, beware of only reviewing the previous iteration's
changes during the review. Whilst recent changes should be the emphasis, a

> review on the entire product, or at least, [a vertical slice](https://en.wikipedia.org/wiki/Vertical_slice) of the product each and every review will hedge against the "disjointed features syndrome".

Author: Will fix. Just for you Curtis!

## Procrastinating Deployment To End Users

Being agile isn't about perfectionism, it is about testing what needs to be
perfect, and discarding things that shouldn't be pursued to perfection.
Part of this mentality is exposing end users to the product being built early
on in the development process. Finding friendly early adopters or testers will
enable a systems-level perspective on the product in a way that the delivery
team may not be able to.

The immediate delivery team of a product is constantly exposed to the product,
they understand it intimately, and are accustomed to its flaws and features.

> Often, it is a new and different perspective that highlight obvious flaws in

Author: Won't fix, highlight is correct as it is in the present tense.

> Often, it is a new and different perspective that highlight obvious flaws in

Reviewer: Singular present tense for highlight is highlights. 

There's only one perspective. Yes Curtis has requested backup. Don't @ me.

> Often, it is a new and different perspective that highlight obvious flaws in

Reviewer: You can either choose to write it "Often new and different perspectives highlight obvious flaws in blah blah blah" or use Curtis' fix.

> Often, it is a new and different perspective that highlight obvious flaws in

Author: OK, will fix :P

a product. This is where useability and experience testing with early adopters
pays off.

## Pursuing the "Extensive Detail" Dichotomy

> When the "disjointed features" syndrome occurs, user stories will often get

Reviewer 2: ```suggestion
When the "disjointed features syndrome" occurs, user stories will often get
```

blamed for not having enough detail. Solutions that are often recommended
during post-mortems or retrospectives is to make user stories more detailed.
Unfortunately, this creates a situation of localised thinking within a user

> story, and more often than not, create situations where the "disjoined feature

Reviewer 2: ```suggestion
story, and more often than not, creates situations where the "disjointed feature
```

syndrome" is amplified rather than reduced.

Localised thinking is the antithesis of systems thinking, and systems thinking

> is an important aspect in hedging against the "Disjointed Features Syndrome".

Reviewer 2: ```suggestion
is an important aspect in hedging against the "disjointed features syndrome".
```
